server/utils/world/initializeWorldDataObject.ts

```diff
@@ -0,0 +1,51 @@
+import { Credentials } from "../../types/Credentials.js";
+import { errorHandler } from "../errorHandler.js";
+import { getDefaultKeyAssetImage } from "../getDefaultKeyAssetImage.js";
+import { DroppedAsset } from "../topiaInit.js";
+
+export const initializeWorldDataObject = async ({ credentials, world }: { credentials: Credentials, world: any }) => {
+  try {
+    let { assetId, urlSlug } = credentials
+    const sceneDropId = credentials.sceneDropId || assetId
+
+    const droppedAsset = await DroppedAsset.get(assetId, urlSlug, { credentials });
+
+    const payload = {
+      itemsCollectedByUser: {},
+      keyAssetId: assetId,
+      lastInteractionDate: new Date(),
+      numberAllowedToCollect: 5,
+      // @ts-ignore
+      questItemImage: droppedAsset.dataObject?.questItemImage || await getDefaultKeyAssetImage(urlSlug),
+      sceneDropId,
+      totalItemsCollected: 0,
+    }
+
+    const lockId = `${sceneDropId}-${new Date(Math.round(new Date().getTime() / 60000) * 60000)}`;
+    if (!world.dataObject?.scenes) {
+      await world.setDataObject(
+        {
+          scenes: {
+            [sceneDropId]: { ...payload },
+          },
+        },
+        { lock: { lockId, releaseLock: true } },
+      );
+    } else if (!world.dataObject?.scenes?.[sceneDropId]) {
+      await world.updateDataObject(
+        {
+          [`scenes.${sceneDropId}`]: { ...payload },
+        },
+        { lock: { lockId, releaseLock: true } },
+      );
+    }
+    return;
+  } catch (error) {
+    errorHandler({
+      error,
+      functionName: "initializeWorldDataObject",
+      message: "Error initializing world data object",
+    });
+    return await world.fetchDataObject();
```

Comment:
This will crash the app for sure, eventually
we shouldn't do awaits inside of catch statements

Reply:
Having an await call inside a catch statement is not inherently bad if done correctly and with caution and intentionality. In this case a user would only land in the catch if the data object has never been initialized AND something goes "wrong" like it's being updated in that same instance by another call. This would only ever happen if multiple clicks occurred for the first time ever at the same time. Once the data object is initialized it would never occur again. Also if it DOES occur and someone falls into the catch because the data object is busy then we do want to refetch so that they have the latest data